sdk/formrecognizer/Azure.AI.FormRecognizer/src/DocumentModelOperationDetails.cs

```diff
@@ -3,39 +3,81 @@
 
 using System;
 using System.Collections.Generic;
-using System.Text.Json;
-using Azure.Core;
 
 namespace Azure.AI.FormRecognizer.DocumentAnalysis
 {
-    [CodeGenModel("GetOperationResponse")]
-    public partial class DocumentModelOperationDetails
+    /// <summary>
+    /// Details about a document model long-running operation.
+    /// </summary>
+    public class DocumentModelOperationDetails
     {
-        // This property is set by the DocumentAnalysisModelFactory when mocking this class.
-        private readonly ResponseError _mockError;
+        internal DocumentModelOperationDetails(GetOperationResponse response)
+            : this(response.OperationId, response.Status, response.PercentCompleted, response.CreatedOn, response.LastUpdatedOn, response.Kind, response.ResourceLocation, response.Tags, response.Error, response.Result)
+        {
+        }
 
-        /// <summary>
-        /// Initializes a new instance of ModelOperation. Used by the <see cref="DocumentAnalysisModelFactory"/>.
-        /// </summary>
-        internal DocumentModelOperationDetails(string operationId, DocumentOperationStatus status, int? percentCompleted, DateTimeOffset createdOn, DateTimeOffset lastUpdatedOn, DocumentOperationKind kind, string resourceLocation, string apiVersion, IReadOnlyDictionary<string, string> tags, ResponseError error, DocumentModelDetails result) : base(operationId, status, percentCompleted, createdOn, lastUpdatedOn, kind, resourceLocation, apiVersion, tags)
+        internal DocumentModelOperationDetails(string operationId, DocumentOperationStatus status, int? percentCompleted, DateTimeOffset createdOn, DateTimeOffset lastUpdatedOn, DocumentOperationKind kind, Uri resourceLocation, IReadOnlyDictionary<string, string> tags, ResponseError error, DocumentModelDetails result)
         {
-            _mockError = error;
+            OperationId = operationId;
+            Status = status;
+            PercentCompleted = percentCompleted;
+            Kind = kind;
+            ResourceLocation = resourceLocation;
+            CreatedOn = createdOn;
+            LastUpdatedOn = lastUpdatedOn;
+            Tags = tags;
+            Error = error;
             Result = result;
         }
 
-        /// <summary> Operation result upon success. </summary>
-        //TODO service is looking into fixing this so it has different return types that we can adapt.
```

Comment:
Currently investigating what this TODO message meant, but not related to this PR.

Reply:
The `Result` property is actually defined as a generic `object` in the swagger specification. We are, however, forcibly parsing it to a `DocumentModelDetails` reference in the SDK. Even though this is the only type of result currently returned by the service, this could change in the future and break the SDK.

Created issue to track this here: https://github.com/Azure/azure-sdk-for-net/issues/30226